### PR TITLE
import youtube urls on desktop v0.2.0

### PR DESCRIPTION
**Pull request by puskuruk**
import-youtube-urls-v0 2 0

Almost re-written. All structrue was changed for being more modular and easier to understand.

### CODE REVIEW

File: desktop/main.js [hunk 1]
```diff
@@ -22,7 +22,8 @@ function createWindow() {
 
   // and load the index.html of the app.
   if (process.env.USE_DEV_SERVER) {
-    mainWindow.loadURL("http://localhost:6001")
+    // USAGE: DEV_SERVER_URL=https://e2a6dfb7.ngrok.io npm run start:desktop:dev
+    mainWindow.loadURL(process.env.DEV_SERVER_URL || "http://localhost:6001")
```
Comment (seveibar) [inline desktop/main.js:26]: nice

File: src/components/ImportFromYoutubeUrls/index.js [hunk 2]
```diff
@@ -1,325 +1,232 @@
-import React, { useState, useEffect } from "react"
+import React, { useState, useEffect, useRef } from "react"
 import { styled } from "@material-ui/core/styles"
 import SimpleDialog from "../SimpleDialog"
-import * as ytdl from "ytdl-core"
```
Comment (seveibar) [inline src/components/ImportFromYoutubeUrls/index.js:4]: should help w/ bundle size

File: src/components/ImportFromYoutubeUrls/index.js [hunk 3]
```diff
@@ -1,325 +1,232 @@
...
-            ? totalProgress.totalFinished
-            : null}
-        </h2>
-      </TotalPercentageWrapper>
-      {/* =======================================* Total Percentage End *======================================= */}
+      {error && <ErrorText>{error}</ErrorText>}
+      {isDownloading && <Progress unitProgress={unitProgress} completedVideoTitles={completedVideoTitles} overallProgress={overallProgress} />}
+      {!isDownloading &&
+        <React.Fragment>
+          <SelectVideoQuality> s
```
Comment (seveibar) [inline src/components/ImportFromYoutubeUrls/index.js:215]: there's a random "s" here

Comment (puskuruk) [reply src/components/ImportFromYoutubeUrls/index.js]: Deleted on newer commits :smile:

File: src/components/ImportFromYoutubeUrls/download-youtube-video.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    // const downloadedPercentage = (percent * 100).toFixed(2)
+    // const downloadedSize = (downloaded / 1024 / 1024).toFixed(2)
+    // const downloadedMBs = (
+    //   downloaded /
+    //   1024 /
+    //   1024
+    // ).toFixed(2)
+    // const totalSize = (total / 1024 / 1024).toFixed(2)
+    // const informationText = `${downloadedPercentage}% downloaded (${downloadedMBs}MB of ${totalSize}MB)\nestimated time left: ${(downloadedMinutes / percent -downloadedMinutes).toFixed(2)}minutes`
```
Comment (seveibar) [inline src/components/ImportFromYoutubeUrls/download-youtube-video.js:50]: Is this supposed to be commented out?

File: src/components/ImportFromYoutubeUrls/get-youtube-video-information.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      progressCallback({
+        progress: (i / (splittedURLsArray.length - 1)) * 100,
+        text: "Inspecting Video Information...",
+      })
+    }
+  }
+  return checkedVideos
+}
+
+export default getYoutubeVideoInformation
```
Comment (seveibar) [inline src/components/ImportFromYoutubeUrls/get-youtube-video-information.js:34]: nice good refactor very clearn

File: src/components/ImportFromYoutubeUrls/split-urls-from-text-area.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  const splittedURLsByNewLines = stringURLs.split("\n")
+  for (const url of splittedURLsByNewLines) {
+    if (url.includes("https://")) {
+      urlsHasHTTPS.push(url)
+    }
+  }
+  return urlsHasHTTPS
+}
+
+export default splitURLsFromTextArea
```
Comment (seveibar) [inline src/components/ImportFromYoutubeUrls/split-urls-from-text-area.js:12]: nice

File: src/components/ImportFromYoutubeUrls/download-youtube-video.js (hunk 4 above)
Comment (seveibar) [reply src/components/ImportFromYoutubeUrls/download-youtube-video.js:50]: I think it's good to merge without this content, easy enough to put back in :)

**Pull request merged**